### PR TITLE
bpo-45516: add protocol description to the Traversable documentation

### PR DESCRIPTION
**Pull request by FFY00**
Signed-off-by: Filipe Laíns <lains@riseup.net>

https://bugs.python.org/issue45516

### CODE REVIEW

File: Lib/importlib/abc.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def read_text(self, encoding=None):
     @abc.abstractmethod
     def is_dir(self) -> bool:
         """
-        Return True if self is a dir
+        Return True if self is a directory
```
Comment (jaraco) [inline Lib/importlib/abc.py]: Let's omit this change and commit it to importlib_resources instead... and it'll make its way here in due time.

**Review by jaraco — APPROVED** [review]

File: Doc/library/importlib.rst [hunk 2]
```diff
@@ -815,6 +815,46 @@ ABC hierarchy::
...
+
+      The base name of this object without any parent references.
+
+    .. abstractmethod:: iterdir()
+
+       Yield Traversable objects in self.
+
+    .. abstractmethod:: is_dir()
+
+       Return True if self is a directtory.
```
Comment (jaraco) [inline Doc/library/importlib.rst]:
```suggestion
       Return True if self is a directory.
```

File: Lib/importlib/abc.py (hunk 1 above)
Comment (FFY00) [reply Lib/importlib/abc.py]:
```suggestion
        Return True if self is a dir
```

File: Doc/library/importlib.rst [hunk 3]
```diff
@@ -815,6 +815,46 @@ ABC hierarchy::
 
     .. versionadded:: 3.9
 
+    .. abstractmethod:: name()
+
+      The base name of this object without any parent references.
```
Comment (jaraco) [inline Doc/library/importlib.rst]:
```suggestion
       The base name of this object without any parent references.
```

**Comment by jaraco** [comment]
> > We can backport this, right?
> 
> Not exactly. Docs aren't synced between CPython and importlib_resources, so including this additional detail is going to require adding a lot more unsynchronized reference material to importlib_resources. It's probably not worth it. Let's just maintain the reference material here.

Wait. Did you mean backport to importlib_resources or backport to Python 3.10/3.9? If the latter, then probably yes.

**Comment by FFY00** [comment]
The later, I wasn't sure sure you maintain a cpython branch.

**Pull request merged**